helpers/persistPostVisitActions.ts

```diff
@@ -175,6 +171,108 @@ const postNoteToBackend = async (
   }
 };
 
+export const persistUnableToEnterPostVisitActions = async (
+  unableToEnterData: Partial<
+    ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]
+  >,
+  processRef: ProcessRef
+): Promise<void> => {
+  if (!unableToEnterData) {
+    return;
+  }
+
+  const unableToEnterMap = processPostVisitActionMap["unableToEnter"];
+  const externalDatabase = await Storage.ExternalContext?.database;
+
+  if (!externalDatabase) {
+    return;
+  }
+  const residents = await externalDatabase.get("residents", processRef);
+  const tenantNames = residents
+    ? residents.tenants.map((tenant) => tenant.fullName).join(", ")
+    : "N/A";
+  const address = residents ? residents.address.join(", ") : "N/A";
+
+  const firstFailedAttempt = unableToEnterData["firstFailedAttempt"];
+  const secondFailedAttempt = unableToEnterData["secondFailedAttempt"];
+  const thirdFailedAttempt = unableToEnterData["thirdFailedAttempt"];
+  const fourthFailedAttempt = unableToEnterData["fourthFailedAttempt"];
+
+  const firstFailedAttemptNotes = firstFailedAttempt
+    ? firstFailedAttempt["notes"] || "N/A"
+    : "N/A";
```

Comment:
I suggest this as a terser way to express this:

```suggestion
  const firstFailedAttemptNotes = (firstFailedAttempt && firstFailedAttempt["notes"]) || "N/A";
```